mods/ra/maps/infiltration/infiltration.lua

```diff
@@ -0,0 +1,380 @@
+		Trigger.Clear(unit, "OnIdle")
+	end
+end
+
+AttackTown = function()
+	Utils.Do(TownAttackers, IdleHunt)
+
+	Trigger.OnRemovedFromWorld(Hospital, function()
+		Utils.Do(TownAttackers, StopHunt)
+	end)
+end
+
+CapOre = function(player)
+	if player.Resources > player.ResourceCapacity * 0.9 then
+		player.Resources = player.ResourceCapacity * 0.8
+	end
+end
+
+NewPatrol = function(actorType, start, waypoints)
+	local guard = Actor.Create(actorType, true, { Owner = soviets, Location = start })
+	guard.Patrol(waypoints, true, Utils.RandomInteger(50, 75))
+end
+
+SetupPatrols = function()
+	Utils.Do(Patrol1, function(patrol1) NewPatrol(patrol1, PatrolPoints1[1], PatrolPoints1) end)
+	Utils.Do(Patrol2, function(patrol2) NewPatrol(patrol2, PatrolPoints1[3], PatrolPoints1) end)
+	Utils.Do(Patrol2, function(patrol3) NewPatrol(patrol3, PatrolPoints3[1], PatrolPoints3) end)
+	Utils.Do(Patrol2, function(patrol4) NewPatrol(patrol4, PatrolPoints4[1], PatrolPoints4) end)
+
+	if Difficulty == "hard" then
+		Utils.Do(Patrol3, function(patrol5) NewPatrol(patrol5, PatrolPoints2[1], PatrolPoints2) end)
+	end
+
+	local checkpoint = { BaseGuardTruckPos.Location }
+	Trigger.OnEnteredFootprint(checkpoint, function(a, id)
+		Trigger.RemoveFootprintTrigger(id)
+		if not BaseGuard.IsDead then
+			BaseGuard.ScriptedMove(BaseGuardMovePos.Location)
+		end
+	end)
+end
+
+ticked = 0
+SecureLabTimer = function()
+	if not timerStarted or labSecured then
+		return
+	end
+
+	if ticked > 0 then
+		UserInterface.SetMissionText("Secure lab in: " .. Utils.FormatTime(ticked), TimerColor)
+		ticked = ticked - 1
+	elseif ticked <= 0 then
+		TimerColor = soviets.Color
+		UserInterface.SetMissionText("The Soviet research laboratory was not secured in time.", TimerColor)
+		SecureLabFailed()
+	end
+end
+
+SovietBaseMaintenanceSetup = function()
+	local sovietbuildings = Utils.Where(Map.NamedActors, function(a)
+		return a.Owner == soviets
+			and a.HasProperty("StartBuildingRepairs") and a.HasProperty("Sell")
+	end)
+
+	-- This includes killed, captured (actor is temporarily removed) and sold.
+	Trigger.OnAllRemovedFromWorld(sovietbuildings, function()
+		Utils.Do(humans, function(player)
+			player.MarkCompletedObjective(destroyBase)
+		end)
+	end)
+
+	Utils.Do(sovietbuildings, function(sovietbuilding)
+		Trigger.OnDamaged(sovietbuilding, function(building)
+			if building.Health < building.MaxHealth * 3/4 then
+				building.StartBuildingRepairs()
+			end
+			if building.Health < building.MaxHealth * 1/4 then
+				building.Sell()
+			end
+		end)
+	end)
+end
+
+CheckPlayerDefeat = function()
+	if not reinforcementsHaveArrived then
+		return
+	end
+
+	Utils.Do(humans, function(player)
+		if player.HasNoRequiredUnits() then
+			player.MarkFailedObjective(destroyBase)
+		end
+	end)
+end
+
+labSecured = false
+CheckLabSecured = function()
+	if not reinforcementsHaveArrived or labSecured then
+		return
+	end
+
+	if player1.HasNoRequiredUnits() and player2 and player2.HasNoRequiredUnits() then
```

Comment:
Make it `if player1.HasNoRequiredUnits() or (player2 and player2.HasNoRequiredUnits()) then`

It won't happen anyway, since the AI doesn't run attacks by itself.